src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -3,26 +3,29 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Buffers;
-using EditorBrowsableAttribute = System.ComponentModel.EditorBrowsableAttribute;
-using EditorBrowsableState = System.ComponentModel.EditorBrowsableState;
 using System.Diagnostics;
-using System.Runtime;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using EditorBrowsableAttribute = System.ComponentModel.EditorBrowsableAttribute;
+using EditorBrowsableState = System.ComponentModel.EditorBrowsableState;
 
 namespace System
 {
+    /// <summary>
+    /// Represents a contiguous region of memory, similar to <see cref="ReadOnlySpan{T}"/>.
+    /// Unlike <see cref="ReadOnlySpan{T}"/>, it is not a byref-like type.
+    /// </summary>
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
     [DebuggerTypeProxy(typeof(MemoryDebugView<>))]
     public readonly struct ReadOnlyMemory<T>
     {
         // NOTE: With the current implementation, Memory<T> and ReadOnlyMemory<T> must have the same layout,
         // as code uses Unsafe.As to cast between them.
 
-        // The highest order bit of _index is used to discern whether _arrayOrOwnedMemory is an array or an owned memory
-        // if (_index >> 31) == 1, object _arrayOrOwnedMemory is an OwnedMemory<T>
-        // else, object _arrayOrOwnedMemory is a T[]
-        private readonly object _arrayOrOwnedMemory;
+        // The highest order bit of _index is used to discern whether _object is an array/string or an owned memory
+        // if (_index >> 31) == 1, _object is an OwnedMemory<T>
+        // else, _object is a T[] or string
+        private readonly object _object;
```

Comment:
This change introduces another difference between ReadOnlyMemory and Memory, semantically, since Memory cannot wrap a string but ReadOnlyMemory can. Can you explain why string has to be wrapped in a ReadOnlyMemory\<char\> only rather than both?

Reply:
Because string is immutable.  Exposing the ability to mutate the string via the `Memory`'s `Span`'s writable ref indexer would be really bad.

Reply:
How is `MemoryMarshal.AsMemory` going to work?

Reply:
I was thinking about that actually last night.
One option would simply be that this is more important, and ditch AsMemory. Another would be making AsMemory instead be just for Memory over bytes. Another would be throwing from AsMemory if it wrapped a string, or making it a Try method. We could also make it work, but then we hit the problem of if it wraps a string, Memory.Span will give you a writable span for a string... maybe that's ok since you went through Marshal and all bets are off?